### ISSUE TITLE
Make TBv2 content choices dynamic

### ISSUE THREAD

**Issue by JeffAbrahamson** [transport-nantes/tn_web#189]
As indicated [here](https://github.com/transport-nantes/tn_web/issues/115#issuecomment-903139055), let's not offer TBv2 types we don't support.  So rather than using an enum and so having to update models on each change, perhaps it makes more sense to have the types be a `DISTINCT()` query on the `TopicBlogTemplate` table.

To do this, we should add a field, the `template_type`, say, which would be "article", "newsletter", etc.  And that way as we add template types, the list will update automatically.

**Comment by Shriukan33** [comment]
For now, precising the item type has no effect really. You can select 'article' or 'newsletter' it wont change the display of the content just yet, because we didn't create templates for that. 

However we already have a `template` field, which serves the purpose you describe. 

What do you want to work differently to be precise ?

**Comment by JeffAbrahamson** [comment]
Yes, the template field is the key, the question here is about the _contents_ of the template field.  And the item type.

You currently (or will soon ;-) have the user choose an item type.  Based on that item type, you offer a choice of templates that satisfy that item type.  No sense offering templates for petitions if the user wants a press release.  (I haven't checked the code.  Since we only have one template at the moment, it's easy to gloss over that we'll later have more.)

But the item type itself is set from an enum.  But I think we should push that list into `TopicBlogTemplate`.  Add an item type there and then just drop the item type in `TopicBlogItem`, instead letting the user choose it on the path to choosing a template (`tb_item.template.item_type`).  Once you choose an item type, you're shown the templates that satisfy that item type.

Make sense?

Wdyt?

**Comment by Shriukan33** [comment]
Makes sense, several templates for same type of content, got it. 

We only have one template for now (content.html) which should be a sort of boilerplate for other templates. 

I think I understand what you mean ! Say we have 2 templates for articles ( ex: `side_image_article.html` and `default_article.html`), selection "article" in item type would leave only those two templates in the "template" field ? 

i have to see if I can change the content of a ChoiceField dynamically but I think I got the concept.

**Comment by JeffAbrahamson** [comment]
Yes, exactly.

If you can't change the type, you can delete the field and make a new field.  After all, there's only a handful of records in existence, you can update them all by hand (or by SQL).

**Comment by Shriukan33** [comment]
tried to get something working, but couldn't find a solution just yet. 
Did a [SO](https://stackoverflow.com/questions/69451723/dynamic-modelchoicefield-on-template) topic, see if someone has an idea.

**Comment by JeffAbrahamson** [comment]
I'm re-opening this:

The commit on this issue is useful, but it doesn't do what we wanted to do here, although it does admittedly do it (usefully) backwards.

When the user selects a content type, you provide a list of available templates.

But the list of available content types is hard coded.  It should only offer content types that actually exist in the database.  (I think you can do that with a static query a la surveys.)  Said differently, you shouldn't be able to choose a content type that leads to no templates being available (unless maybe you animate a group of laughing clowns).
